hs_core/tests/api/rest/test_resource_types.py

```diff
@@ -0,0 +1,23 @@
+import json
+
+from rest_framework.test import APIClient
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from hs_core.hydroshare.utils import get_resource_types
+
+
+class TestResourceTypes(APITestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_resource_typelist(self):
+        resource_types = set([t.__name__ for t in get_resource_types()])
+
+        response = self.client.get('/hsapi/resourceTypes/', format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        content = json.loads(response.content)
+        rest_resource_types = set([t['resource_type'] for t in content])
+
+        self.assertEqual(resource_types, rest_resource_types)
```

Comment:
@ironfroggy There is documentation in the [wiki](https://github.com/hydroshare/hydroshare/wiki/HydroShare-REST-API), but not documentation of supported resource types.  I will add these.

@pkdash Do we have a published list of supported resource types?  I don't think so, but I want to make sure I do not duplicate what might already be out there.